### PR TITLE
Fix/rack attack configuration

### PR DESCRIPTION
**Pull request by Quentinchampenois**
#### Description

Update Rack attack configuration, aim is to cleanup the log files generated and have a better overview

#### Tasks

- [x] Update throttle rules
- [x] Apply rails secrets for configuration
- [x] Allow to disable Rack::Attack
- [x] Add specs
- [x] Create a custom 429 Error page 
- [ ] Configure Allow2Ban
- [x] Update logs format
```bash 
W, [2022-09-27T10:13:25.989874 #19108]  WARN -- : [d49f682a-029c-412d-96af-3a07f62dbb77] {"ip"=>"127.0.0.1", "path"=>"/", "get"=>{"locale"=>"en"}, "post"=>{}, "host"=>"localhost", "referer"=>"http://localhost:3000/users/sign_in?locale=en"}

```
 
#### How to test

Run `ENABLE_RACK_ATTACK=1 THROTTLING_MAX_REQUESTS=3 bundle exec rails s`, you should be limited at 3 requests

#### Screenshots
Screenshot 2022-09-27 at 14 10 32

### CODE REVIEW

File: config/initializers/rack_attack.rb [hunk 1]
```diff
@@ -1,15 +1,120 @@
...
+    border-bottom-right-radius: 4px;
+    border-top-color: #DADADA;
+    color: #666;
+    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
+  }
+  </style>
+</head>
+
+<body class='rails-default-error-page'>
+  <!-- This file lives in public/422.html -->
```
Comment (moustachu) [inline config/initializers/rack_attack.rb:107]: remove HTML comment :wink:

File: config/initializers/rack_attack.rb [hunk 2]
```diff
@@ -1,15 +1,120 @@
...
+
+  rack_logger.warn("[#{request_uuid}] #{params}")
+
+  [429, { "Content-Type" => "text/html" }, [html_template(until_period)]]
+end
+
+Rack::Attack.throttle("req/ip",
+                      limit: Rails.application.secrets.decidim[:throttling_max_requests],
+                      period: Rails.application.secrets.decidim[:throttling_period]) do |req|
+  req.ip unless req.path.start_with?("/assets") || req.path.start_with?("/rails/active_storage")
```
Comment (moustachu) [inline config/initializers/rack_attack.rb:45]: Change `/assets` path to `/decidim-packs`

File: config/initializers/rack_attack.rb [hunk 3]
```diff
@@ -1,15 +1,120 @@
...
+
+# By default use the memory store for inspecting requests
+# Better to use MemCached or Redis in production mode
+Rack::Attack.cache.store = ActiveSupport::Cache::MemoryStore.new if !ENV["MEMCACHEDCLOUD_SERVERS"] || Rails.env.test?
+
+Rack::Attack.throttled_responder = lambda do |request|
+  match_data = request.env["rack.attack.match_data"]
+  now = match_data[:epoch_time]
+  # throttle_time = match_data[:period]
+  until_period = I18n.l(Time.zone.at(now + 60), format: :decidim_short)
```
Comment (moustachu) [inline config/initializers/rack_attack.rb:17]: Use `match_data[:period]` instead of hard coded value `60`

**Review by moustachu — CHANGES_REQUESTED** [review]
We should remove the original throttle configuration from decidim-core.
https://github.com/OpenSourcePolitics/decidim-app/blob/fix/rack-attack-double-throttle/config/initializers/rack_attack.rb#L6

_It's not possible to overwrite an existing configuration with a "throttle block", you need to remove it directly from the configuration hash_

**Review by armandfardeau — APPROVED** [review]

**Review by moustachu — APPROVED** [review]

**Review by paulinebessoles — APPROVED** [review]

**Pull request merged**